src/Raven.Server/Rachis/Leader.cs

```diff
@@ -732,6 +739,10 @@ private void EmptyQueue()
             }
             catch (Exception e)
             {
+                if (_running.IsRaised() == false)
+                {
+                    e = lostLeadershipException;
```

Comment:
Probably a bad idea. We need to create a new instance with the inner exception set.

Reply:
done.